### PR TITLE
Couple convective cloud to radiation

### PR DESCRIPTION
**Pull request by dustinswales**
Opened on behalf of @Qingfu-Liu, replacing #8 
This PR contains adds the convective cloud water (liquid water + ice water) in the calculations of Cloud Water Path and Ice Water Path for radiation cloud properties. 
For RRTMG radiation, only one file changed: radiation_clouds.f (code changed by Ruiyu Sun)
For RRTMGP radiation, one file has changed and there is an upstream change needed in NOAA-EMC/fv3atm for a new interstitial variable.

This will change the baselines for any RT using the Thompson MP scheme
[TestingAndEvaluation_PR9.pdf](https://github.com/ufs-community/ccpp-physics/files/9678514/TestingAndEvaluation_PR9.pdf)

### CODE REVIEW

**Review by RuiyuSun — APPROVED** [review]
the change in radiation_clouds routine looks fine

**Comment by RuiyuSun** [comment]
Some of the results from the change in the radiation_clouds routine can be seen 
 here. [https://docs.google.com/presentation/d/14OJnTJZkO4EI-I3UQXLeRd0_AIVUOK9S/edit#slide=id.p11](url)

**Comment by ChunxiZhang-NOAA** [comment]
@dustinswales Could you add Ruiyu's results into the description of this PR?

File: physics/radiation_clouds.f [hunk 1]
```diff
@@ -2216,8 +2216,10 @@ subroutine progcld_thompson_wsm6                                  &
 
         do k = 1, NLAY-1
           do i = 1, IX
-            cwp(i,k) = max(0.0, clw(i,k,ntcw) * gfac * delp(i,k))
-            cip(i,k) = max(0.0, clw(i,k,ntiw) * gfac * delp(i,k))
+            cwp(i,k) = max(0.0, (clw(i,k,ntcw)+cnvw(i,k)*
```
Comment (mzhangw) [inline physics/radiation_clouds.f:2215]: I recommend documenting this code change at Line 2215: "...considering the impact of convective clouds."

**Comment by dustinswales** [comment]
> @dustinswales Could you add Ruiyu's results into the description of this PR?

@RuiyuSun @ChunxiZhang-NOAA I don't think that link is working.

**Comment by Qingfu-Liu** [comment]
The link is (need copy/paste):
https://docs.google.com/presentation/d/14OJnTJZkO4EI-I3UQXLeRd0_AIVUOK9S/edit#slide=id.p11

File: physics/radiation_clouds.f (hunk 1 above)
Comment (dustinswales) [reply physics/radiation_clouds.f:2215]: @mzhangw Great idea. Done

**Review by ChunxiZhang-NOAA — APPROVED** [review]
Looks good.

**Comment by dustinswales** [comment]
@mzhangw @ChunxiZhang-NOAA
I forgot that we needed a fix for [NCAR issue #957](https://github.com/NCAR/ccpp-physics/issues/957), and agreed to put it into this PR. Can you both review [this addition](https://github.com/ufs-community/ccpp-physics/pull/9/commits/c635c0503e3909eb9e0ccfd08f4e7a275ec8679d)?
@Qingfu-Liu @yangfanglin

File: physics/GFS_rrtmgp_cloud_mp.F90 [hunk 2]
```diff
@@ -292,6 +297,14 @@ subroutine GFS_rrtmgp_cloud_mp_run(nCol, nLev, nTracers, ncnd, i_cldliq, i_cldic
        endif
     endif
 
+    ! Instantaneous 2D (max-in-column) cloud fraction
+    do iCol = 1, nCol
+       cldfra2d(iCol) = 0._kind_phys
+       do iLay = 1, nLev-1
```
Comment (ChunxiZhang-NOAA) [inline physics/GFS_rrtmgp_cloud_mp.F90:304]: An alternative way is using cldfra2d(iCol) = maxval(cld_fra(iCol,1:nLev-1)). Probably this method is more efficient?

Comment (dustinswales) [reply physics/GFS_rrtmgp_cloud_mp.F90:304]: I prefer your suggestion over the current coding. Not sure of the history, I just used the [same code as is in RRTMG](https://github.com/NCAR/ccpp-physics/blob/main/physics/GFS_rrtmg_pre.F90#L1016)

File: physics/GFS_rrtmgp_cloud_mp.F90 [hunk 3]
```diff
@@ -706,16 +722,15 @@ subroutine cloud_mp_thompson(nCol, nLev, nTracers, ncnd, i_cldliq, i_cldice, i_c
...
     integer :: iCol,iLay,l
 
     ! Cloud condensate
     cld_condensate(1:nCol,1:nLev,1) = tracer(1:nCol,1:nLev,i_cldliq)     ! -liquid water
     cld_condensate(1:nCol,1:nLev,2) = tracer(1:nCol,1:nLev,i_cldice)     ! -ice water
-    cld_condensate(1:nCol,1:nLev,3) = tracer(1:nCol,1:nLev,i_cldrain)    ! -rain water
-    cld_condensate(1:nCol,1:nLev,4) = tracer(1:nCol,1:nLev,i_cldsnow) + &! -snow + grapuel
-                                      tracer(1:nCol,1:nLev,i_cldgrpl)
+    cld_condensate(1:nCol,1:nLev,3) = tracer(1:nCol,1:nLev,i_cldrain)    ! -rain hydrometeors
+    cld_condensate(1:nCol,1:nLev,4) = tracer(1:nCol,1:nLev,i_cldsnow)    ! -snow hydrometeors
```
Comment (grantfirl) [inline physics/GFS_rrtmgp_cloud_mp.F90:734]: I didn't see anywhere in the description; why was graupel removed here?

**Review by grantfirl — APPROVED** [review]

**Comment by ChunxiZhang-NOAA** [comment]
In radiation_clouds.f, Ruiyu's new recommendation is to change line 3909 from clwt = 1.0e-10 * (plyr(i,k)*0.001) to clwt = 1.0e-6 * (plyr(i,k)*0.001). @dustinswales Could you modify it in this PR?

File: physics/radiation_clouds.f [hunk 4]
```diff
@@ -3906,7 +3906,7 @@ subroutine cloud_fraction_mass_flx_2                              &
         clwmin = 0.0
         do k = 1, NLAY-1
         do i = 1, IX
-          clwt = 1.0e-10 * (plyr(i,k)*0.001)
+          clwt = 1.0e-6 * (plyr(i,k)*0.001)
```
Comment (dustinswales) [inline physics/radiation_clouds.f:3907]: @ChunxiZhang-NOAA Done

Comment (ChunxiZhang-NOAA) [reply physics/radiation_clouds.f:3907]: @dustinswales Thanks!

Comment (RuiyuSun) [reply physics/radiation_clouds.f:3907]: Looks good.

**Comment by grantfirl** [comment]
@dustinswales Please update this to the latest ufs-dev branch when you get a chance in order to start testing.

**Comment by grantfirl** [comment]
@dustinswales Since the CI changes were merged, this needs to be updated to ufs/dev again before testing.

**Pull request merged**